src/xkbcomp/parser.y

```diff
@@ -59,25 +58,32 @@ _xkbcommon_error(struct parser_param *param, const char *msg)
 }
 
 static bool
-resolve_keysym(struct parser_param *param, const char *name, xkb_keysym_t *sym_rtrn)
+resolve_keysym(struct parser_param *param, struct sval name, xkb_keysym_t *sym_rtrn)
 {
     xkb_keysym_t sym;
 
-    if (!name || istreq(name, "any") || istreq(name, "nosymbol")) {
+    if (isvaleq(name, SVAL_LIT("any")) || isvaleq(name, SVAL_LIT("nosymbol"))) {
         *sym_rtrn = XKB_KEY_NoSymbol;
         return true;
     }
 
-    if (istreq(name, "none") || istreq(name, "voidsymbol")) {
+    if (isvaleq(name, SVAL_LIT("none")) || isvaleq(name, SVAL_LIT("voidsymbol"))) {
         *sym_rtrn = XKB_KEY_VoidSymbol;
         return true;
     }
 
-    sym = xkb_keysym_from_name(name, XKB_KEYSYM_NO_FLAGS);
+    char buf[256];
```

Comment:
This is a bit dubious, but I think we can rely on a max keysym name size, @wismill WDYT?

Ideally we'd have a variant of xkb_keysym_from_name which takes a buf+len instead of C string, then we could avoid this copy entirely (and would also be nice for FFI bindings in languages which don't use C strings), but it's not immediately straightforward so I went with this copy for now.